File: cukcuk clone/app/src/main/java/vn/com/misa/cukcukstarterclone/data/source/local/dao/OrderDao.java
package vn.com.misa.cukcukstarterclone.data.source.local.dao;

import android.content.ContentValues;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;

import java.util.ArrayList;
import java.util.List;

import vn.com.misa.cukcukstarterclone.data.model.Cart;
import vn.com.misa.cukcukstarterclone.data.model.CartItem;
import vn.com.misa.cukcukstarterclone.data.model.DetailsReport;
import vn.com.misa.cukcukstarterclone.data.model.MenuItem;
import vn.com.misa.cukcukstarterclone.data.model.Order;
import vn.com.misa.cukcukstarterclone.data.source.local.database.AppDatabase;

/**
 * @created_by KhanhNQ on 28-Jan-2021.
 */
public class OrderDao implements IOrderDao {

    private final SQLiteDatabase database;

    private OrderDao(AppDatabase appDatabase) {
        this.database = appDatabase.getWritableDatabase();
    }

    private static OrderDao instance = null;

    public static OrderDao getInstance(AppDatabase appDatabase) {
        if (null == instance) {
            instance = new OrderDao(appDatabase);
        }
        return instance;
    }

    @Override
    public List<Order> getAllOrders() {
        Cursor cursor = database.query(Order.TABLE_NAME, null, null, null, null, null, null);
        List<Order> list = new ArrayList<>();
        if (cursor.moveToFirst()) {
            while (!cursor.isAfterLast()) {
                list.add(new Order(cursor));
                cursor.moveToNext();
            }
        }
        cursor.close();
        return list;
    }

    @Override
    public List<Order> getOrdersByDate(String date) {
        String from = date.split("to")[0];
        String to = date.split("to")[1] + " 23:59:59";
        String selection = Order.CREATED_AT + " >= ? AND " + Order.CREATED_AT + " <= ?";
        String[] selectionArg = {from, to};
        Cursor cursor = database.query(Order.TABLE_NAME, null, selection, selectionArg, null, null, null);
        List<Order> orders = new ArrayList<>();
        if (cursor.moveToFirst()) {
            while (!cursor.isAfterLast()) {
                orders.add(new Order(cursor));
                cursor.moveToNext();
            }
        }
        cursor.close();
        return orders;
    }

    @Override
    public Order getOrderById(String id) {
        String selection = Order.ID + " =?";
        String[] selectionArg = {id};
        Cursor cursor = database.query(Order.TABLE_NAME, null, selection, selectionArg, null, null, null);
        cursor.moveToFirst();
        Order item = new Order(cursor);
        cursor.close();
        return item;
    }

    @Override
    public boolean addNewOrder(Order newOrder) {
        ContentValues contentValues = newOrder.getContentValues();
        return database.insert(Order.TABLE_NAME, null, contentValues) > 0;
    }
}

File: app/src/main/java/kma/longhoang/beta/dao/OrderDAO.kt
package kma.longhoang.beta.dao

import android.annotation.SuppressLint
import android.content.ContentValues
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import kma.longhoang.beta.database.AppDatabase
import kma.longhoang.beta.model.OrderModel
import kma.longhoang.beta.model.ProductModel

class OrderDAO private constructor(appDatabase: AppDatabase) {
    private val database: SQLiteDatabase = appDatabase.writableDatabase

    @SuppressLint("Recycle")
    fun getAllOrder(): MutableList<OrderModel> {
        val cursor: Cursor = database.rawQuery("SELECT * FROM tb_order", null)
        val list = mutableListOf<OrderModel>()
        if (cursor.moveToFirst()) {
            while (!cursor.isAfterLast) {
                list.add(OrderModel(cursor))
                cursor.moveToNext()
            }
        }
        cursor.close();
        return list
    }

    @SuppressLint("Recycle")
    fun getOrderById(billId: Int): MutableList<OrderModel> {
        val cursor: Cursor =
            database.rawQuery("SELECT * FROM tb_order WHERE billId = $billId", null)
        val list = mutableListOf<OrderModel>()
        if (cursor.moveToFirst()) {
            while (!cursor.isAfterLast) {
                list.add(OrderModel(cursor))
                cursor.moveToNext()
            }
        }
        cursor.close()
        return list
    }

    fun addOrder(order: OrderModel, billId: Int): Boolean {
        val value = ContentValues()
        value.put("name", order.name)
        value.put("code", order.code)
        value.put("color", order.color?.name)
        value.put("price", order.price)
        value.put("amount", order.amount)
        value.put("billId", billId)
        return database.insert("tb_order", null, value) > 0
    }

    companion object {
        private var instance: OrderDAO? = null
        fun getInstance(appDatabase: AppDatabase): OrderDAO? {
            if (null == instance) {
                instance = OrderDAO(appDatabase)
            }
            return instance
        }
    }
}